src/client/scenes/Game.ts

```diff
@@ -139,33 +140,30 @@ export default class Game extends Phaser.Scene {
   // function to add new player to the otherPlayer group
   private handlePlayerJoined(newPlayer: IPlayer, key: string) {
     if (!this.otherPlayers) {
-      this.otherPlayers = this.physics.add.group({ classType: OtherPlayer })
+      this.otherPlayers = new Map<string, OtherPlayer>()
     }
-    this.otherPlayers.get(newPlayer.x, newPlayer.y, 'player', key) as OtherPlayer
+    const otherPlayer = this.add.otherPlayer(newPlayer.x, newPlayer.y, 'player')
```

Comment:
I'm not sure about this but I believe using a group in phaser has better performance than adding these one by one, so I was expecting you keep the group and just add another Map to keep track of all the OtherPlayer:

```ts
private otherPlayerMap = new Map<string, OtherPlayer>()
private otherPlayers: Phaser.Physics.Arcade.Group = this.physics.add.group({ classType: OtherPlayer })

// handlePlayerJoined()
const otherPlayer = this.otherPlayers.get(...) as OtherPlayer
this.otherPlayerMap.set(id, otherPlayer)

// handlePlayerLeft(id: string)
const otherPlayer = this.otherPlayerMap.get(id)
if (!otherPlayer) return
otherPlayer.destroy() // don't know if still need this
this.otherPlayers.remove(otherPlayer, true, true)
this.otherPlayerMap.delete(id)
```

Reply:
I agree. I actually found out later that having a group for other players can be very handy for handling events (e.g. onOverlap) for them.

```typescript
otherPlayer.destroy() // don't know if still need this
```
This is not needed because having the following means Phaser will remove this child from scene (first ```true```) and destroy it (second ```true```)
```typescript
this.otherPlayers.remove(otherPlayer, true, true)
```